src/main/java/com/shopify/minishopify/controllers/ProductController.java

```diff
@@ -0,0 +1,105 @@
+package com.shopify.minishopify.controllers;
+
+import com.shopify.minishopify.model.Product;
+import com.shopify.minishopify.model.Shop;
+import com.shopify.minishopify.repository.ProductRepository;
+import com.shopify.minishopify.repository.ShopRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.validation.Valid;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api")
+public class ProductController {
+    private ProductRepository productRepository;
+    private ShopRepository shopRepository;
+
+    public ProductController(ProductRepository productRepository, ShopRepository shopRepository) {
+        this.productRepository = productRepository;
+        this.shopRepository = shopRepository;
+    }
+
+    @GetMapping("/shop/{id}/products")
+    public Iterable<Product> getAllProducts(@PathVariable int id) {
+        return productRepository.findByShopId(id);
+    }
+
+    @GetMapping("/products/{id}")
```

Comment:
For cleaner code, maybe change routing to single "product" here

Reply:
/products/id means from a list of product return the one with id